src/snowflake/cli/_plugins/workspace/entity_commands.py

```diff
@@ -0,0 +1,102 @@
+from snowflake.cli.api.commands.snow_typer import SnowTyperFactory
+from snowflake.cli.api.entities.common import EntityActions
+from snowflake.cli.api.output.types import CommandResult, MessageResult
+from snowflake.cli.api.project.definition_manager import DefinitionManager
+from snowflake.cli.api.project.schemas.entities.entities import (
+    Entity,
+    EntityModel,
+    v2_entity_model_to_entity_map,
+    v2_entity_model_types_map,
+)
+
+
+class EntityCommandGroup(SnowTyperFactory):
+    target_id: str
+    model_type: EntityModel
+    entity_type: Entity
+
+    def __init__(self, target_id: str, model_type_str: str):
+        super().__init__(
+            name=f"@{target_id}",
+            help=f"Commands to interact with the {target_id} entity defined in {DefinitionManager.BASE_DEFINITION_FILENAME}.",
+        )
+        self.target_id = target_id
+        self.model_type_str = model_type_str
+        self.model_type = v2_entity_model_types_map[model_type_str]
+        self.entity_type = v2_entity_model_to_entity_map[self.model_type]
+
+    @property
+    def supported_actions(self):
+        return sorted(
+            [action for action in EntityActions if self.entity_type.supports(action)]
+        )
+
+    def register_commands(self):
+        for action in self.supported_actions:
+            verb = action.value.split("action_")[1]
+            action_callable = getattr(self.entity_type, action)
+
+            # TODO: get args for action and turn into typer options
+            # TODO: what message result are we returning? do we throw them away for multi-step actions (i.e deps?)
+            # TODO: how do we know if a command needs connection?
+
+            @self.command(verb)
+            @with_project_definition()
+            def _action_executor(**options) -> CommandResult:
+                cli_context = get_cli_context()
+                ws = WorkspaceManager(
+                    project_definition=cli_context.project_definition,
+                    project_root=cli_context.project_root,
+                )
+                # entity = ws.get_entity(self.target_id)
+                ws.perform_action(self.target_id, action)
+                return MessageResult(
+                    f"Successfully performed {verb} on {self.target_id}."
+                )
+
+            _action_executor.__doc__ = action_callable.__doc__
+
+
+def generate_entity_commands(
+    ws: SnowTyperFactory, project_root: Path | str | None = None
+):
+    """
+    Introspect the current snowflake.yml file, generating @<id> command groups
+    for each found entity. Throws a fatal ClickException if templating is used
+    in the basic definition of entities (i.e. the type discriminator field).
+    """
+    dm = DefinitionManager(str(project_root) if project_root is not None else None)
+
+    if not dm.has_definition_file:
+        return
+
+    root = dm.unrendered_project_definition
+    for (target_id, model) in root.entities.items():
```

Comment:
Should we consider `model.get_command_group(target_id)` that would encapsulate L100-102? WDYT?

Reply:
I'm not sure I quite understand the suggestion. Can you give a bit more detail?